copado/main/default/classes/Promotion.cls

```diff
@@ -0,0 +1,52 @@
+public class Promotion extends DomainBuilder {
+
+    public Promotion(Project prj, Release rls, Credential source, Environment sourceEnv, Credential dest, Environment destEnv) {
+        super(copado__Promotion__c.SObjectType);
+        setParent(copado__Promotion__c.copado__Project__c, prj);
```

Comment:
Are they all independant? Why not take the Project from the Release the Environment from the Credential etc?

Reply:
I am doing it in other constructors, I wanted to give the possibility of just passing project, for example, because the release is optional. On the other hand, I think we can get rid of the environment and just retrieve it from the credential